### PR TITLE
Camera upload configuration helper

### LINKED ISSUE #123: add possibility to choose custom folder for camera-auto-upload
It would be great if you could choose a directory on your device which is used for the camera-auto-upload feature.
It's very common to store pictures on extSDCard, if the device has one. It should be possible to auto-upload from SDCard.
Even devices without extSDCard will profit from this option since some peope use extraordinarily camera apps which sometimes store photos in other directorys than the standard DCIM folder.

### LINKED ISSUE #140: get photo path from external SD card
Seadroid 1.2 only supports uploading photos from primary external storage not the secondary external storage. That\`s to say.  
- [x] if the device only has one emulated external storage (on-board flash), camera upload service can scan photo directories on it correctly.
- [x] if the device only has a removable storage (like SD card), camera upload service can scan photo directories correctly as well. So far so good.
- [x] if the device both has a emulated external storage and a removable storage, camera upload service only scans photo directories on the emulated external storage (primary storage), but a majority of users  save their photos to the 32G or bigger size removable SD card (secondary storage) which will be ignored by Seadroid 1.2. That\`s where problem comes in.

### LINKED ISSUE #151: let user choose which photo directories they want to upload
preview

### LINKED ISSUE #158: let user choose a photo upload directory on SD card

### PR DESCRIPTION
**Pull request by Logan676**
1. let user choose one or more custom local directories from internal storage or removable SD card.
2. let user choose one library on remote server
3. upload via WiFi or data plan
4. upload photos only or videos included

fix #123 
fix #140 
fix #151 
fix #158 

preview
how_to_upload
what_to_upload
local_directories
remote_library
settings_menu

Note that, this patch was inspired by [JamsMusicPlayer](https://github.com/psaravan/JamsMusicPlayer).

### CODE REVIEW

**Comment by schokonsti** [comment]
I really like this feature!

**Comment by Logan676** [comment]
@lins05 this patch is almost there I think, please help review. Thanks.

**Comment by Logan676** [comment]
@lins05 please help review. Thanks.

**Comment by Logan676** [comment]
@lins05 I have fixed you review tickets
1. hide `upload to parent folder` button at the root of list
2. nav back to previous page (if has) when press back button
3. show selected upload directories in Settings
4. other fixes

I have tested it according to this [test case](https://github.com/haiwen/seadroid/wiki/Camera-Upload-Test-Case). 
Please help to review. Thanks.

File: src/com/seafile/seadroid2/cameraupload/CloudLibraryAdapter.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+/**
+ * Cloud library adapter
+ */
+public class CloudLibraryAdapter extends ReposAdapter {
+
+    protected LinkedHashMap<SeafRepo, Boolean> repos = new LinkedHashMap<SeafRepo, Boolean>();
```
Comment (lins05) [inline src/com/seafile/seadroid2/cameraupload/CloudLibraryAdapter.java]: This map is a `repo -> is_repo_selected` map, but when choosing the target repo for camera upload, there would only be one library that would be selected, so it doesn't make sense to use a hash map here.

File: src/com/seafile/seadroid2/cameraupload/CameraUploadConfigActivity.java [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+        public Fragment getItem(int position) {
+
+            if (isChooseLibPage) {
+                switch (position) {
+                    case 0:
+                        mCloudLibFragment = new CloudLibraryFragment();
+                        return mCloudLibFragment;
+                    default:
+                        return null;
+                }
```
Comment (lins05) [inline src/com/seafile/seadroid2/cameraupload/CameraUploadConfigActivity.java]: The above ~6 lines inside the `if` block can be simplified as `return position == 0 ? new CloudLibraryFragment() : null`

**Comment by lins05** [comment]
The code mostly looks good to me. Please squash all the commits into one commit.

**Pull request merged**